### PR TITLE
Add character set for vietnamese phoneme

### PR DESCRIPTION
**Pull request by vndee**
I have modified the character set for entirely Vietnamese support with the espeak-ng phoneme. As a result, it will be better for Vietnamese text-to-speech pre-processing.

For example, it contains number and t̪ symbols in Vietnamese phonemes:

```
Raw text: Tiếng Việt, cũng gọi là tiếng Việt Nam hay Việt ngữ là ngôn ngữ của người Việt
Phonemes: |t̪|ˈiɛɜ|ŋ| v|ˈiɛ6|t̪| k|ˈu5|ŋ| ɣ|ˈɔ6|j| l|ˌaː2| t̪|ˈiɛɜ|ŋ| v|ˈiɛ6|t̪| n|ˈaː1|m| h|ˈa1|j| v|ˈiɛ6|t̪| ŋ|ˈy5| l|ˌaː2| ŋ|ˈo1|n| ŋ|ˈy5| k|ˌuə4| ŋ|ˈyə2|j| v|ˈiɛ6|t̪|
```

### CODE REVIEW

File: TTS/tts/utils/text/phonemizers/espeak_wrapper.py [hunk 1]
```diff
@@ -228,8 +228,9 @@ def is_available(cls):
...
 
-    e = ESpeak(language="en-us", keep_puncs=False)
-    print("`" + e.phonemize("hello how are you today?") + "`")
+    e = ESpeak(language="vi", keep_puncs=False)
+    print("`" + e.phonemize("Tiếng Việt, cũng gọi là tiếng Việt Nam hay Việt ngữ là ngôn ngữ của người Việt.") + "`")
 
+    print("-" * 30)
     e = ESpeak(language="en-us", keep_puncs=True)
-    print("`" + e.phonemize("hello how are you today?") + "`")
+    print("`" + e.phonemize("Tiếng Việt, cũng gọi là tiếng Việt Nam hay Việt ngữ là ngôn ngữ của người Việt.") + "`")
```
Comment (p0p4k) [inline TTS/tts/utils/text/phonemizers/espeak_wrapper.py:236]: Maybe use the default English sentence for this second example.

File: TTS/tts/utils/text/characters.py [hunk 2]
```diff
@@ -28,11 +28,20 @@ def parse_symbols():
 # Phonemes definition (All IPA characters)
 _vowels = "iyɨʉɯuɪʏʊeøɘəɵɤoɛœɜɞʌɔæɐaɶɑɒᵻ"
 _non_pulmonic_consonants = "ʘɓǀɗǃʄǂɠǁʛ"
-_pulmonic_consonants = "pbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟ"
+_pulmonic_consonants = "pbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟt̪"
 _suprasegmentals = "ˈˌːˑ"
 _other_symbols = "ʍwɥʜʢʡɕʑɺɧʲ"
 _diacrilics = "ɚ˞ɫ"
-_phonemes = _vowels + _non_pulmonic_consonants + _pulmonic_consonants + _suprasegmentals + _other_symbols + _diacrilics
+_numbers = "123456789"
```
Comment (p0p4k) [inline TTS/tts/utils/text/characters.py:35]: Aren't numbers supposed to be taken care of in the `text_cleaners`?

File: TTS/tts/utils/text/phonemizers/espeak_wrapper.py (hunk 1 above)
Comment (vndee) [reply TTS/tts/utils/text/phonemizers/espeak_wrapper.py]: Thanks for your reply, I changed the second example to the default English sentence.

File: TTS/tts/utils/text/characters.py (hunk 2 above)
Comment (vndee) [reply TTS/tts/utils/text/characters.py:35]: To my understanding, the text_cleaners module is applied right before the phonemizer. So it does not affect the phoneme sequence. These _numbers are just the notion of Vietnamese phonetics, which was produced by espeak-ng.
https://github.com/coqui-ai/TTS/blob/d6ad9a05b40e9f3aa876ecfa1c0d23339f373df4/TTS/tts/utils/text/tokenizer.py#L107

Comment (erogol) [reply TTS/tts/utils/text/characters.py:35]: @vndee what you mean by "just the notion of Vietnamese phonetics"? Are they more than numbers in Vietnamese?

**Comment by erogol** [comment]
Isn't it possible to solve this by providing additional phonemes in the config? I think this change would break the released models since it changes the order of phonemes.

**Comment by HTInnovation** [comment]
> Thanks for the sparklesPRsparkles. Isn't it possible to solve this by providing additional phonemes in the config? I think this change would break the released models since it changes the order of phonemes.

Ahh, thanks for your concern. I will check if I can bring it to the config option.

**Comment by vndee** [comment]
> Thanks for the sparklesPRsparkles. Isn't it possible to solve this by providing additional phonemes in the config? I think this change would break the released models since it changes the order of phonemes.

Ahh, thanks for your concern. I will check if I can bring it to the config option.

File: TTS/tts/utils/text/characters.py (hunk 2 above)
Comment (longcon123) [reply TTS/tts/utils/text/characters.py]: > @vndee what you mean by "just the notion of Vietnamese phonetics"? Are they more than numbers in Vietnamese?
No only 1-9! I think his means is the IPA phonetic of Vietnamese which produced by the espeak-ng including numbers..
ie..'Xin chào các bạn' => s|i|n|1| |c|a|w|2| |k|a|k|5| |b|a|n|6| |. 
That why he add the _numbers.

Comment (longcon123) [reply TTS/tts/utils/text/characters.py]: @vndee I think it only contains numbers from 1-6

**Comment by vndee** [comment]
Since I figure out that we can add extra phoneme characters via `characters` param in the config, so I close this PR.

**Pull request closed without merging**